File: flink-job-example.py
from kafka import KafkaConsumer
from pyflink.datastream import StreamExecutionEnvironment
from pyflink.datastream.connectors import FlinkKafkaConsumer
from pyflink.common.serialization import SimpleStringSchema

client_id = 'example-flink-job'
shared_topic = 'flink-topic'

def process_message(value):
    # Example processing logic
    return value.upper()

def run_flink_job():
    env = StreamExecutionEnvironment.get_execution_environment()

    kafka_consumer = FlinkKafkaConsumer(
        topics=shared_topic,
        deserialization_schema=SimpleStringSchema(),
        properties={'bootstrap.servers': 'localhost:9092', 'group.id': 'flink-group'}
    )

    data_stream = env.add_source(kafka_consumer)

    processed_stream = data_stream.map(process_message)

    processed_stream.print()

    env.execute('Flink Kafka Consumer Job')

if __name__ == '__main__':
    run_flink_job()

File: flink-job-example.js
const { Kafka } = require('kafkajs');
const { Flink } = require('flink');

const clientId = 'example-flink-job';
const sharedTopic = 'flink-topic';

const kafka = new Kafka({
  brokers: ['localhost:9092'],
  clientId: clientId,
});

const consumer = kafka.consumer({ groupId: 'flink-group' });

async function runFlinkJob() {
  await consumer.connect();
  await consumer.subscribe({ topic: sharedTopic, fromBeginning: true });

  console.log('Flink job started. Press Ctrl+C to exit.');

  await consumer.run({
    eachMessage: async ({ topic, partition, message }) => {
      const value = message.value.toString();
      console.log(`Received message: ${value}`);

      // Process the message using Flink
      const processedValue = processMessageWithFlink(value);
      console.log(`Processed message: ${processedValue}`);
    },
  });
}

function processMessageWithFlink(value) {
  // Example Flink processing logic
  return value.toUpperCase();
}

async function shutdown() {
  console.log('Shutting down Flink job...');
  await consumer.disconnect();
  process.exit(0);
}

runFlinkJob().catch(console.error);

process.on('SIGINT', shutdown);
